tests/lib/rules/init-declarations.js

```diff
@@ -0,0 +1,313 @@
+                {
+                    message: "Variable 'c' should be initialized on declaration.",
+                    type: "VariableDeclaration"
+                }
+            ]
+        },
+        {
+            code: "var foo = bar = 2;",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "never"],
+            errors: [
+                {
+                    message: "Variable 'foo' should not be initialized on declaration.",
+                    type: "VariableDeclaration"
+                }
+            ]
+        },
+        {
+            code: "var foo = true;",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "never"],
+            errors: [
+                {
+                    message: "Variable 'foo' should not be initialized on declaration.",
+                    type: "VariableDeclaration"
+                }
+            ]
+        },
+        {
+            code: "var foo, bar = 5, baz = 3;",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "never"],
+            errors: [
+                {
+                    message: "Variable 'bar' should not be initialized on declaration.",
+                    type: "VariableDeclaration"
+                },
+                {
+                    message: "Variable 'baz' should not be initialized on declaration.",
+                    type: "VariableDeclaration"
+                }
+            ]
+        },
+        {
+            code: "function foo() { var foo; var bar = foo; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "never"],
+            errors: [
+                {
+                    message: "Variable 'bar' should not be initialized on declaration.",
+                    type: "VariableDeclaration"
+                }
+            ]
+        },
+        {
+            code: "let a = 1;",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "never"],
+            errors: [
+                {
+                    message: "Variable 'a' should not be initialized on declaration.",
+                    type: "VariableDeclaration"
+                }
+            ]
+        },
+        {
+            code: "function foo() { let a = 'foo', b; if (a) { let c, d; } }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "never"],
+            errors: [
+                {
+                    message: "Variable 'a' should not be initialized on declaration.",
+                    type: "VariableDeclaration"
+                }
+            ]
+        },
+        {
+            code: "function foo() { let a; const b = false; var c = 1; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "never"],
```

Comment:
Can you use `options` instead of args? It's the same except you don't need the 2 (which is never used anywhere). We are trying to deprecate args.